gateway-ha/src/test/java/io/trino/gateway/ha/router/TestQueryCountBasedRouter.java

```diff
@@ -0,0 +1,250 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.gateway.ha.router;
+
+import com.google.common.collect.ImmutableList;
+import io.trino.gateway.ha.clustermonitor.ClusterStats;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotEquals;
+
+public class TestQueryCountBasedRouter
+{
+    static final String BACKEND_URL_1 = "http://c1";
+    static final String BACKEND_URL_2 = "http://c2";
+    static final String BACKEND_URL_3 = "http://c3";
+    static final String BACKEND_URL_4 = "http://c4";
+    static final String BACKEND_URL_5 = "http://c5";
+    static final String BACKEND_URL_UNHEALTHY = "http://c-unhealthy";
+
+    static final int LEAST_QUEUED_COUNT = 1;
+    static final int SAME_QUERY_COUNT = 5;
+    QueryCountBasedRouter queryCountBasedRouter;
+    ImmutableList<ClusterStats> clusters;
+
+    TestQueryCountBasedRouter()
+    {
+        queryCountBasedRouter = null;
+        clusters = null;
+    }
+
+    // Helper function to generate the ClusterStat list
+    private static ImmutableList getClusterStatsList(String routingGroup)
```

Comment:
Please fix the warning about raw usage of parameterized list class in this class. 
```suggestion
    private static List<ClusterStats> getClusterStatsList(String routingGroup)
```

Reply:
This one line change is there with last commit, but not sure that is the complete fix. How do I see the details on the errors (or original error).  I don't see it on my local build with `./mvnw test-compile`; also didn't find any warning here -
https://github.com/trinodb/trino-gateway/actions/runs/8249053834/job/22560656476?pr=98